### PR TITLE
pd: heartbeat record the time interval of report

### PR DESCRIPTION
**Pull request by nolouch**
Do not let this logic be processed in PD, and use to calculate some flow information and denoising

### CODE REVIEW

**Comment by siddontang** [comment]
can this improve the balance? any result?

**Comment by overvenus** [comment]
Please resolve conflicts and merge the master, the failed test has been fixed in the master.

File: src/pd/client.rs [hunk 1]
```diff
@@ -301,9 +301,12 @@ impl PdClient for RpcClient {
         req.set_keys_read(region_stat.read_bytes);
         req.set_approximate_size(region_stat.approximate_size);
 
-        let now = SystemTime::now();
-        let ts = now.duration_since(UNIX_EPOCH).unwrap().as_secs();
-        req.set_timestamp(ts);
+        let ts = SystemTime::now()
```
Comment (siddontang) [inline src/pd/client.rs]: can you extract a function for these? although this is simple, it takes many lines.

**Comment by siddontang** [comment]
any update? @nolouch

File: src/util/time.rs [hunk 2]
```diff
@@ -43,6 +43,15 @@ pub fn duration_to_nanos(d: Duration) -> u64 {
     d.as_secs() * 1_000_000_000 + nanos
 }
 
+/// Get the current timestamp in seconds.
+#[inline]
+pub fn time_now_sec() -> u64 {
+    SystemTime::now()
```
Comment (overvenus) [inline src/util/time.rs]: I worry about `SystemTime`s may vary in cluster if the ntp is not enabled.

Comment (nolouch) [reply src/util/time.rs]: It doesn't need to compare with others, only compare himself. if the leader change the interval will not in a reasonable range, we will ignore it.

**Comment by disksing** [comment]
LGTM.

**Comment by Connor1996** [comment]
It changes original field `timestamp` which pd uses, will it cause compatibility issue?

**Comment by nolouch** [comment]
@Connor1996  yep, already fixed in kvproto.

File: src/util/time.rs (hunk 2 above)
Comment (siddontang) [reply src/util/time.rs]: maybe you can add another coarse time and use it to reduce the system call load.

Comment (nolouch) [reply src/util/time.rs]: @siddontang in code https://github.com/rust-lang/rust/blob/1eab9c5a1b85db4ca850114814a7df3e289fcd67/src/libstd/sys/unix/time.rs#L353-L364.  it uses the `clock_gettime` function to get time, that exported by the vDSO.

**Pull request merged**